### PR TITLE
Pylons - Separate settings for in-game and zeus

### PR DESCRIPTION
**Pull request by 654wak654**
**When merged this pull request will:**
- Switch pylons to CBA Settings
- Split the "enabled" setting into "enabled for zeus" and "enabled for ammo trucks" settings (#6336).

### CODE REVIEW

File: addons/pylons/ACE_Settings.hpp [hunk 1]
```diff
@@ -1,46 +0,0 @@
-class ACE_Settings {
```
Comment (mharis001) [inline addons/pylons/ACE_Settings.hpp]: `movedToSQF = 1` for settings moved to use `CBA_settings_fnc_init`?

File: addons/pylons/XEH_preInit.sqf [hunk 2]
```diff
@@ -6,4 +6,67 @@ PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
 PREP_RECOMPILE_END;
 
+[
```
Comment (PabstMirror) [inline addons/pylons/XEH_preInit.sqf]: We have been putting settings into a file called initSettings.sqf
and then `#include "initSettings.sqf"`

File: addons/pylons/XEH_postInit.sqf [hunk 3]
```diff
@@ -1,57 +1,53 @@
...
-    if (isServer) then {
-        GVAR(currentAircraftNamespace) = true call CBA_fnc_createNamespace;
-        publicVariable QGVAR(currentAircraftNamespace);
-
-        addMissionEventHandler ["HandleDisconnect", LINKFUNC(handleDisconnect)];
+                private _vehicles = nearestObjects [_target, ["Air", "LandVehicle", "Slingload_base_F", "ReammoBox_F"], GVAR(searchDistance) + 10];
+                private _filter = ["transportAmmo", QEGVAR(rearm,defaultSupply)] select (["ace_rearm"] call EFUNC(common,isModLoaded));
+                private _rearmVehicles = {(getNumber (configFile >> "CfgVehicles" >> typeOf _x >> _filter)) > 0} count _vehicles;
+
+                (_rearmVehicles > 0 && {[ace_player, _target] call FUNC(canConfigurePylons)} && {GVAR(enabledFromAmmoTrucks)})
```
Comment (PabstMirror) [inline addons/pylons/XEH_postInit.sqf]: I would add `if (!GVAR(enabledFromAmmoTrucks)) exitWith {false}` at the top of this condition to save a lot of time if it's disabled

File: addons/pylons/XEH_postInit.sqf [hunk 4]
```diff
@@ -1,57 +1,53 @@
...
 }] call CBA_fnc_addEventHandler;
+
+if (isServer) then {
+    GVAR(currentAircraftNamespace) = true call CBA_fnc_createNamespace;
+    publicVariable QGVAR(currentAircraftNamespace);
+
+    addMissionEventHandler ["HandleDisconnect", LINKFUNC(handleDisconnect)];
+};
+
+GVAR(searchDistanceSqr) = GVAR(searchDistance) ^ 2;
```
Comment (PabstMirror) [inline addons/pylons/XEH_postInit.sqf]: needs to be back inside the `ace_settingsInitialized` event, or use settings changed code

File: addons/pylons/initSettings.sqf [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+] call CBA_Settings_fnc_init;
+
+[
+    QGVAR(searchDistance),
+    "SLIDER",
+    [LSTRING(SearchDistance), LSTRING(SearchDistance_description)],
+    LSTRING(Category_Pylons),
+    [5, 50, 15, 0],
+    true,
+	{
```
Comment (PabstMirror) [inline addons/pylons/initSettings.sqf]: tabs

**Comment by 654wak654** [comment]
This closes #6336 btw @PabstMirror

**Pull request merged**